### ISSUE TITLE
JWT, JWK and JWA separation.

### ISSUE THREAD

**Issue by lolgear** [yourkarma/JWT#174]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/yourkarma/JWT/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/yourkarma/JWT#documentation)
* [x] I have searched for a similar issue in the [project](https://github.com/yourkarma/JWT/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Although JWT implementation is not a rocket science, it has become huge. I would like to separate implementations of key extraction ( JWK ), jWT handling and algorithms implementations ( JWA ).

So, at the end we would have different targets for testing and it will reduces API and elements coupling.
